client/www/pages/docs/storage.md

```diff
@@ -5,287 +5,743 @@ title: Storage
+            Upload Image
+          </button>
+        </div>
+      )}
+    </div>
+  );
+}
 
-Let's imagine you have an `images` namespace you use to store the file metadata of your images. You can use this to keep track of the expiration time of all your file URLs, and then refresh them accordingly.
+function ImageGrid({ images }: { images: InstantFile[] }) {
+  const [deletingIds, setDeletingIds] = React.useState<Set<string>>(new Set());
 
-```tsx
-// Simple component to upload and display image files
-function App() {
-  const { data } = db.useQuery({ images: {} });
+  const handleDelete = async (image: InstantFile) => {
+    setDeletingIds((prev) => new Set([...prev, image.id]));
 
-  const upload = async (files: FileList) => {
-    const file = files[0];
-    const pathname = file.name; // or whatever custom file path you'd like
-    const expiresAt = Date.now() + 7 * 24 * 60 * 60 * 1000; // 7 days from now
-    const isSuccess = await db.storage.upload(pathname, file);
-    const cachedUrl = await db.storage.getDownloadUrl(pathname);
+    await deleteImage(image);
 
-    db.transact(tx.images[id()].update({ cachedUrl, pathname, expiresAt }));
-  };
+    setDeletingIds((prev) => {
+      prev.delete(image.id);
+      return prev;
+    });
+  }
 
   return (
-    <div>
-      <input type="file" onChange={(e) => upload(e.target.files)} />
-      {data.images.map((image) => (
-        <ImageViewer key={image.id} image={image} />
-      ))}
+    <div className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-5 w-full max-w-6xl">
+      {images.map((image) => {
+        const isDeleting = deletingIds.has(image.id);
+        return (
+          <div key={image.id} className="border border-gray-300 rounded-lg overflow-hidden">
+            <div className="relative">
+              {/* $files entities come with a `url` property */}
+              <img src={image.url} alt={image.path} className="w-full h-64 object-cover" />
+              {isDeleting && (
+                <div className="absolute inset-0 bg-black bg-opacity-50 flex items-center justify-center">
+                  <div className="w-8 h-8 border-2 border-t-2 border-gray-200 border-t-white rounded-full animate-spin"></div>
+                </div>
+              )}
+            </div>
+
+            <div className="p-3 flex justify-between items-center bg-white">
+              <span>{image.path}</span>
+              <span onClick={() => handleDelete(image)} className="cursor-pointer text-gray-300 px-1">
+                𝘟
+              </span>
+            </div>
+          </div>
+        )
+      })}
     </div>
   );
 }
+
+export default App;
 ```
 
-Then, in your `ImageViewer` component, you can use the `cachedUrl` by default, and handle the expiration when necessary:
+Make sure to update this line with the app id in your `.env` file.
```

Comment:
If they have a .env file, perhaps it makes sense to simply use `process.env.NEXT_PUBLIC_INSTANT_APP_ID` in the example?